### PR TITLE
Clarify C++ standard in documentation

### PR DESCRIPTION
**Pull request by stotko**
- Replace `[[maybe_unused]]` attribute with `(void)...` to avoid using C++17 features
- Clarify allowed C++ standard in contributor guide
- Explicitly mention requirement of a C++14 compiler in building documentation
- Add note about `Filament`'s compiler requirements when building it from source

---
This change is [Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3384)

### CODE REVIEW

File: cpp/open3d/t/geometry/Image.cpp [hunk 1]
```diff
@@ -105,7 +105,7 @@ Image Image::To(core::Dtype dtype,
             {core::Dtype::UInt16},  {core::Dtype::Int32},
             {core::Dtype::Float32}, {core::Dtype::Float64}};
 
-    [[maybe_unused]] double scale = 1.0;
+    double scale = 1.0;
```
Comment (ssheorey) [inline cpp/open3d/t/geometry/Image.cpp:108]: This variable is unused on ARM (no IPP or NPP for supporting this function), so there's a compiler error without this attribute.

**Review by yxlao — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3384)* status: all files reviewed, 1 unresolved discussion (waiting on @errissa, @prewettg, and @ssheorey)

___
*[cpp/open3d/t/geometry/Image.cpp, line 108 at r1](https://reviewable.io/reviews/intel-isl/open3d/3384#-MZsQGRiga3r9Dsdyc17-r1-108:-MZtQTOY1MEUXeNpIV4-:b-zb5qae) ([raw file](https://github.com/intel-isl/open3d/blob/04e9af13a6e838fdbaae3c6f8d2d0bc6d045fd27/cpp/open3d/t/geometry/Image.cpp#L108)):*
<details><summary><i>Previously, ssheorey wrote…</i></summary><blockquote>

This variable is unused on ARM (no IPP or NPP for supporting this function), so there's a compiler error without this attribute.
</blockquote></details>

we could also `(void) scale;` to suppress the warning

**Review by prewettg — APPROVED** [review]
> Reviewed 3 of 3 files at r1.
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3384)* status: all files reviewed, 1 unresolved discussion (waiting on @errissa and @ssheorey)

**Review by errissa — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3384)* status: all files reviewed, 1 unresolved discussion (waiting on @ssheorey)

___
*[cpp/open3d/t/geometry/Image.cpp, line 108 at r1](https://reviewable.io/reviews/intel-isl/open3d/3384#-MZsQGRiga3r9Dsdyc17-r1-108:-MZuqzkKDqFzYAsTT2-2:bfbbpl5) ([raw file](https://github.com/intel-isl/open3d/blob/04e9af13a6e838fdbaae3c6f8d2d0bc6d045fd27/cpp/open3d/t/geometry/Image.cpp#L108)):*
<details><summary><i>Previously, yxlao (Yixing Lao) wrote…</i></summary><blockquote>

we could also `(void) scale;` to suppress the warning 
</blockquote></details>

I think `[[maybe_unused]]` is more clear. Either way, a comment explaining the need for the attribute is warranted.

**Review by stotko — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/intel-isl/open3d/3384)* status: 2 of 3 files reviewed, 1 unresolved discussion (waiting on @prewettg, @ssheorey, and @yxlao)

___
*[cpp/open3d/t/geometry/Image.cpp, line 108 at r1](https://reviewable.io/reviews/intel-isl/open3d/3384#-MZsQGRiga3r9Dsdyc17-r1-108:-MZvOPDAAwQedTOpvc-A:b-9dyj4m) ([raw file](https://github.com/intel-isl/open3d/blob/04e9af13a6e838fdbaae3c6f8d2d0bc6d045fd27/cpp/open3d/t/geometry/Image.cpp#L108)):*
<details><summary><i>Previously, errissa (Rene) wrote…</i></summary><blockquote>

I think `[[maybe_unused]]` is more clear. Either way, a comment explaining the need for the attribute is warranted.
</blockquote></details>

Thanks for pointing this out. I went with `(void)scale` and adding a comment to it. Technically we are not allowed to use `[[maybe_unused]]` even though many compilers silently accept that.

**Review by ssheorey — APPROVED** [review]

**Pull request merged**